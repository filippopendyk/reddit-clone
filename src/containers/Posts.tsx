import { useAppSelector } from "../hooks";

type Props = {
    category: string | undefined; 
}

const Posts: React.FC<Props> = ({category}) => {

    return (
        <h1>This is Posts container about {category}</h1>
    )
}

export default Posts;