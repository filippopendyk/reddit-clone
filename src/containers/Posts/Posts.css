.posts-container{
    overflow-y: scroll;
    height: 100%;
}