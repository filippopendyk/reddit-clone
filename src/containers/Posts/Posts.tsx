import { useAppDispatch, useAppSelector } from "../../hooks";
import { useEffect } from "react";
import LoadingComp from "../../components/LoadingComp/LoadingComp";
import ErrorComp from "../../components/ErrorComp/ErrorComp";
import PostsDisplayer from "../../components/PostsDisplayer/PostsDisplayer";

type Props = {
    data: {}[];
    isLoading: boolean;
    error: string | null;
}

const Posts: React.FC<Props> = ({isLoading, error, data}) => {

    if(isLoading){
        return <LoadingComp/>
    }

    if(error){
        return <ErrorComp error={error}/>
    }

    return (
        <PostsDisplayer posts={data}/>
    )
}

export default Posts;