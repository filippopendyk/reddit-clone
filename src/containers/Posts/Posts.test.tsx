import { screen, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Posts from "./Posts";
import renderer from 'react-test-renderer';

describe('Posts Container', () => {
    it('Renders correctly', () => {
        const tree = renderer
        .create(<Posts isLoading={false} error={null} data={[]}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});