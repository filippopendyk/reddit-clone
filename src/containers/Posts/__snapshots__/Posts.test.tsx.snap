// Vitest Snapshot v1

exports[`Posts Container > Renders correctly 1`] = `
<h4>
  This is filter posts component
</h4>
`;
